### ISSUE TITLE
Change or allow customisation of date formats (eg ISO8601) in web UI

### ISSUE THREAD

**Issue by jedd** [hashicorp/nomad#18539]
### Proposal

New configuration item to specify date / time format for web UI

This would improve upon the (small) customisation offered via the language/region setting for the browser being used.

In the UI there are currently (1.6.2) these variations on US format, eg:

```
Sep 20 21:44:26 +1000   (Job list of recent allocation)
Sep 20, '23 21:46:33 +1000  (Allocation list of tasks)
```

The Clients / {client} / Overview page has two sections, each using one of the above date formats.

I think the USA is the only place the short-text version of the month is shown before the numeric day of month, but in any case, the formatting is hard to parse for non-Americans.

I note #3806  raised the same issue, but the solution addressed only the CLI  date formatting.

I would expect an additional setting under the ui block  is the logical place - https://developer.hashicorp.com/nomad/docs/configuration/ui

Either something similar to linux `date` command's FORMAT options to provide full customisation, or even just the ability to set ISO860 (YYYY-MM-DD hh:mm:ss tz/offset) would probably be sufficient and satisfactory.

### Use-cases

As above, use case is to improve usability for non-US users, who are not comfortable with either of the two date formats mentioned above.

### Attempted Solutions

n/a

**Comment by tgross** [comment]
Hi @jedd! That seems totally reasonable, but I'm not quite sure off the top of my head how we'd surface that to the UI (which is a SPA). I'll mark this for roadmapping and further discussion though.
